module,execution_time (sec),success,timestamp,error,error_short
cowidev.vax.batch.argentina,4.33,True,2022-07-12T07:00:15,,
cowidev.vax.batch.australia,2.45,True,2022-07-12T07:00:13,,
cowidev.vax.batch.austria,,,2022-07-12T07:00:10,,
cowidev.vax.batch.belgium,3.21,True,2022-07-12T07:00:14,,
cowidev.vax.batch.bolivia,2.35,True,2022-07-12T07:00:13,,
cowidev.vax.batch.canada,5.3,True,2022-07-12T07:00:16,,
cowidev.vax.batch.chile,1.14,True,2022-07-12T07:00:14,,
cowidev.vax.batch.czechia,4.17,True,2022-07-12T07:00:17,,
cowidev.vax.batch.denmark,4.0,True,2022-07-12T07:00:18,,
cowidev.vax.batch.ecdc,60.09,False,2022-07-12T07:04:25,"Traceback (most recent call last):
  File ""/mnt/owid_live_covid/covid-19-data/scripts/venv/lib/python3.9/site-packages/urllib3/connection.py"", line 174, in _new_conn
    conn = connection.create_connection(
  File ""/mnt/owid_live_covid/covid-19-data/scripts/venv/lib/python3.9/site-packages/urllib3/util/connection.py"", line 96, in create_connection
    raise err
  File ""/mnt/owid_live_covid/covid-19-data/scripts/venv/lib/python3.9/site-packages/urllib3/util/connection.py"", line 86, in create_connection
    sock.connect(sa)
socket.timeout: timed out

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File ""/mnt/owid_live_covid/covid-19-data/scripts/venv/lib/python3.9/site-packages/urllib3/connectionpool.py"", line 699, in urlopen
    httplib_response = self._make_request(
  File ""/mnt/owid_live_covid/covid-19-data/scripts/venv/lib/python3.9/site-packages/urllib3/connectionpool.py"", line 382, in _make_request
    self._validate_conn(conn)
  File ""/mnt/owid_live_covid/covid-19-data/scripts/venv/lib/python3.9/site-packages/urllib3/connectionpool.py"", line 1010, in _validate_conn
    conn.connect()
  File ""/mnt/owid_live_covid/covid-19-data/scripts/venv/lib/python3.9/site-packages/urllib3/connection.py"", line 358, in connect
    conn = self._new_conn()
  File ""/mnt/owid_live_covid/covid-19-data/scripts/venv/lib/python3.9/site-packages/urllib3/connection.py"", line 179, in _new_conn
    raise ConnectTimeoutError(
urllib3.exceptions.ConnectTimeoutError: (<urllib3.connection.HTTPSConnection object at 0x7f14be298340>, 'Connection to opendata.ecdc.europa.eu timed out. (connect timeout=30)')

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File ""/mnt/owid_live_covid/covid-19-data/scripts/venv/lib/python3.9/site-packages/requests/adapters.py"", line 440, in send
    resp = conn.urlopen(
  File ""/mnt/owid_live_covid/covid-19-data/scripts/venv/lib/python3.9/site-packages/urllib3/connectionpool.py"", line 755, in urlopen
    retries = retries.increment(
  File ""/mnt/owid_live_covid/covid-19-data/scripts/venv/lib/python3.9/site-packages/urllib3/util/retry.py"", line 574, in increment
    raise MaxRetryError(_pool, url, error or ResponseError(cause))
urllib3.exceptions.MaxRetryError: HTTPSConnectionPool(host='opendata.ecdc.europa.eu', port=443): Max retries exceeded with url: /covid19/vaccine_tracker/csv/data.csv (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x7f14be298340>, 'Connection to opendata.ecdc.europa.eu timed out. (connect timeout=30)'))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File ""/mnt/owid_live_covid/covid-19-data/scripts/src/cowidev/cmd/commons/get.py"", line 47, in run
    module.main()
  File ""/mnt/owid_live_covid/covid-19-data/scripts/src/cowidev/vax/batch/ecdc.py"", line 506, in main
    ECDC().export()
  File ""/mnt/owid_live_covid/covid-19-data/scripts/src/cowidev/vax/batch/ecdc.py"", line 496, in export
    df = self.read().pipe(self.pipe_base)
  File ""/mnt/owid_live_covid/covid-19-data/scripts/src/cowidev/vax/batch/ecdc.py"", line 111, in read
    df = read_csv_from_url(self.source_url, timeout=30)
  File ""/mnt/owid_live_covid/covid-19-data/scripts/src/cowidev/utils/web/download.py"", line 39, in read_csv_from_url
    download_file_from_url(url, tmp.name, timeout=timeout, verify=verify, ciphers_low=ciphers_low)
  File ""/mnt/owid_live_covid/covid-19-data/scripts/src/cowidev/utils/web/download.py"", line 52, in download_file_from_url
    r = requests.get(url, stream=True, timeout=timeout, verify=verify)
  File ""/mnt/owid_live_covid/covid-19-data/scripts/venv/lib/python3.9/site-packages/requests/api.py"", line 75, in get
    return request('get', url, params=params, **kwargs)
  File ""/mnt/owid_live_covid/covid-19-data/scripts/venv/lib/python3.9/site-packages/requests/api.py"", line 61, in request
    return session.request(method=method, url=url, **kwargs)
  File ""/mnt/owid_live_covid/covid-19-data/scripts/venv/lib/python3.9/site-packages/requests/sessions.py"", line 529, in request
    resp = self.send(prep, **send_kwargs)
  File ""/mnt/owid_live_covid/covid-19-data/scripts/venv/lib/python3.9/site-packages/requests/sessions.py"", line 645, in send
    r = adapter.send(request, **kwargs)
  File ""/mnt/owid_live_covid/covid-19-data/scripts/venv/lib/python3.9/site-packages/requests/adapters.py"", line 507, in send
    raise ConnectTimeout(e, request=request)
requests.exceptions.ConnectTimeout: HTTPSConnectionPool(host='opendata.ecdc.europa.eu', port=443): Max retries exceeded with url: /covid19/vaccine_tracker/csv/data.csv (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x7f14be298340>, 'Connection to opendata.ecdc.europa.eu timed out. (connect timeout=30)'))
","HTTPSConnectionPool(host='opendata.ecdc.europa.eu', port=443): Max retries exceeded with url: /covid19/vaccine_tracker/csv/data.csv (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x7f14be298340>, 'Connection to opendata.ecdc.europa.eu timed out. (connect timeout=30)'))"
cowidev.vax.batch.ecuador,1.95,True,2022-07-12T07:00:17,,
cowidev.vax.batch.estonia,1.05,True,2022-07-12T07:00:17,,
cowidev.vax.batch.france,0.5,True,2022-07-12T07:00:17,,
cowidev.vax.batch.germany,1.73,True,2022-07-12T07:00:18,,
cowidev.vax.batch.greece,2.68,True,2022-07-12T07:00:20,,
cowidev.vax.batch.hong_kong,12.44,True,2022-07-12T07:00:30,,
cowidev.vax.batch.indonesia,,,2022-07-12T07:00:18,,
cowidev.vax.batch.ireland,0.51,True,2022-07-12T07:00:18,,
cowidev.vax.batch.israel,10.88,True,2022-07-12T07:00:29,,
cowidev.vax.batch.italy,12.31,True,2022-07-12T07:00:31,,
cowidev.vax.batch.japan,72.28,True,2022-07-12T07:01:32,,
cowidev.vax.batch.jersey,1.61,True,2022-07-12T07:00:31,,
cowidev.vax.batch.latvia,28.07,True,2022-07-12T07:00:58,,
cowidev.vax.batch.lithuania,1.87,True,2022-07-12T07:00:32,,
cowidev.vax.batch.luxembourg,2.0,True,2022-07-12T07:00:33,,
cowidev.vax.batch.malaysia,2.19,True,2022-07-12T07:00:35,,
cowidev.vax.batch.malta,1.32,True,2022-07-12T07:00:34,,
cowidev.vax.batch.netherlands,,,2022-07-12T07:00:34,,
cowidev.vax.batch.new_zealand,36.96,True,2022-07-12T07:01:11,,
cowidev.vax.batch.norway,1.8,True,2022-07-12T07:00:36,,
cowidev.vax.batch.peru,3.82,True,2022-07-12T07:00:40,,
cowidev.vax.batch.portugal,,,2022-07-12T07:00:40,,
cowidev.vax.batch.romania,32.23,True,2022-07-12T07:01:13,,
cowidev.vax.batch.saudi_arabia,5.79,True,2022-07-12T07:01:03,,
cowidev.vax.batch.singapore,2.23,True,2022-07-12T07:01:06,,
cowidev.vax.batch.slovakia,67.08,True,2022-07-12T07:02:13,,
cowidev.vax.batch.slovenia,1.13,True,2022-07-12T07:01:12,,
cowidev.vax.batch.south_korea,43.99,True,2022-07-12T07:01:56,,
cowidev.vax.batch.spc,3.15,False,2022-07-12T07:04:28,"Traceback (most recent call last):
  File ""/mnt/owid_live_covid/covid-19-data/scripts/src/cowidev/cmd/commons/get.py"", line 47, in run
    module.main()
  File ""/mnt/owid_live_covid/covid-19-data/scripts/src/cowidev/vax/batch/spc.py"", line 220, in main
    SPC().export()
  File ""/mnt/owid_live_covid/covid-19-data/scripts/src/cowidev/vax/batch/spc.py"", line 214, in export
    data = self.read()
  File ""/mnt/owid_live_covid/covid-19-data/scripts/src/cowidev/vax/batch/spc.py"", line 59, in read
    return self.parse_data(data)
  File ""/mnt/owid_live_covid/covid-19-data/scripts/src/cowidev/vax/batch/spc.py"", line 73, in parse_data
    return self._build_df_list(vaccination_data)
  File ""/mnt/owid_live_covid/covid-19-data/scripts/src/cowidev/vax/batch/spc.py"", line 103, in _build_df_list
    data[k] = self._build_df(v, k)
  File ""/mnt/owid_live_covid/covid-19-data/scripts/src/cowidev/vax/batch/spc.py"", line 135, in _build_df
    df = df.pipe(make_monotonic)
  File ""/mnt/owid_live_covid/covid-19-data/scripts/venv/lib/python3.9/site-packages/pandas/core/generic.py"", line 5430, in pipe
    return com.pipe(self, func, *args, **kwargs)
  File ""/mnt/owid_live_covid/covid-19-data/scripts/venv/lib/python3.9/site-packages/pandas/core/common.py"", line 471, in pipe
    return func(obj, *args, **kwargs)
  File ""/mnt/owid_live_covid/covid-19-data/scripts/src/cowidev/vax/utils/utils.py"", line 44, in make_monotonic
    raise Exception(
Exception: 50 rows have been removed. That is more than maximum allowed (10) by make_monotonic() - check the data. Check 
          date  ...  vaccine
9   2021-07-13  ...      NaN
10  2021-07-20  ...      NaN
11  2021-07-27  ...      NaN
12  2021-08-02  ...      NaN
13  2021-08-09  ...      NaN
14  2021-08-16  ...      NaN
15  2021-08-23  ...      NaN
16  2021-08-31  ...      NaN
17  2021-09-07  ...      NaN
18  2021-09-14  ...      NaN
19  2021-09-21  ...      NaN
20  2021-09-28  ...      NaN
21  2021-10-05  ...      NaN
22  2021-10-12  ...      NaN
23  2021-10-18  ...      NaN
24  2021-10-25  ...      NaN
25  2021-11-01  ...      NaN
26  2021-11-08  ...      NaN
27  2021-11-15  ...      NaN
28  2021-11-22  ...      NaN
29  2021-11-30  ...      NaN
30  2021-12-06  ...      NaN
31  2021-12-13  ...      NaN
32  2021-12-20  ...      NaN
33  2022-01-04  ...      NaN
34  2022-01-11  ...      NaN
35  2022-01-18  ...      NaN
36  2022-01-24  ...      NaN
37  2022-01-31  ...      NaN
38  2022-02-07  ...      NaN
39  2022-02-14  ...      NaN
40  2022-02-21  ...      NaN
41  2022-02-28  ...      NaN
42  2022-03-07  ...      NaN
43  2022-03-14  ...      NaN
44  2022-03-21  ...      NaN
45  2022-03-28  ...      NaN
46  2022-04-04  ...      NaN
47  2022-04-11  ...      NaN
48  2022-04-18  ...      NaN
49  2022-04-25  ...      NaN
50  2022-05-02  ...      NaN
51  2022-05-09  ...      NaN
52  2022-05-16  ...      NaN
53  2022-05-23  ...      NaN
54  2022-05-30  ...      NaN
55  2022-06-06  ...      NaN
56  2022-06-13  ...      NaN
57  2022-06-20  ...      NaN
58  2022-06-27  ...      NaN

[50 rows x 8 columns]
","50 rows have been removed. That is more than maximum allowed (10) by make_monotonic() - check the data. Check 
          date  ...  vaccine
9   2021-07-13  ...      NaN
10  2021-07-20  ...      NaN
11  2021-07-27  ...      NaN
12  2021-08-02  ...      NaN
13  2021-08-09  ...      NaN
14  2021-08-16  ...      NaN
15  2021-08-23  ...      NaN
16  2021-08-31  ...      NaN
17  2021-09-07  ...      NaN
18  2021-09-14  ...      NaN
19  2021-09-21  ...      NaN
20  2021-09-28  ...      NaN
21  2021-10-05  ...      NaN
22  2021-10-12  ...      NaN
23  2021-10-18  ...      NaN
24  2021-10-25  ...      NaN
25  2021-11-01  ...      NaN
26  2021-11-08  ...      NaN
27  2021-11-15  ...      NaN
28  2021-11-22  ...      NaN
29  2021-11-30  ...      NaN
30  2021-12-06  ...      NaN
31  2021-12-13  ...      NaN
32  2021-12-20  ...      NaN
33  2022-01-04  ...      NaN
34  2022-01-11  ...      NaN
35  2022-01-18  ...      NaN
36  2022-01-24  ...      NaN
37  2022-01-31  ...      NaN
38  2022-02-07  ...      NaN
39  2022-02-14  ...      NaN
40  2022-02-21  ...      NaN
41  2022-02-28  ...      NaN
42  2022-03-07  ...      NaN
43  2022-03-14  ...      NaN
44  2022-03-21  ...      NaN
45  2022-03-28  ...      NaN
46  2022-04-04  ...      NaN
47  2022-04-11  ...      NaN
48  2022-04-18  ...      NaN
49  2022-04-25  ...      NaN
50  2022-05-02  ...      NaN
51  2022-05-09  ...      NaN
52  2022-05-16  ...      NaN
53  2022-05-23  ...      NaN
54  2022-05-30  ...      NaN
55  2022-06-06  ...      NaN
56  2022-06-13  ...      NaN
57  2022-06-20  ...      NaN
58  2022-06-27  ...      NaN

[50 rows x 8 columns]"
cowidev.vax.batch.sweden,18.26,True,2022-07-12T07:01:33,,
cowidev.vax.batch.switzerland,39.76,True,2022-07-12T07:01:59,,
cowidev.vax.batch.trinidad_and_tobago,2.18,True,2022-07-12T07:01:34,,
cowidev.vax.batch.ukraine,,,2022-07-12T07:01:33,,
cowidev.vax.batch.united_kingdom,8.88,True,2022-07-12T07:01:42,,
cowidev.vax.batch.united_states,45.78,True,2022-07-12T07:02:20,,
cowidev.vax.batch.uruguay,4.35,True,2022-07-12T07:01:46,,
cowidev.vax.batch.zimbabwe,0.65,True,2022-07-12T07:01:47,,
cowidev.vax.incremental.africacdc,,,2022-07-12T07:01:47,,
cowidev.vax.incremental.antigua_barbuda,4.88,True,2022-07-12T07:01:52,,
cowidev.vax.incremental.aruba,1.52,True,2022-07-12T07:01:53,,
cowidev.vax.incremental.azerbaijan,14.96,True,2022-07-12T07:02:08,,
cowidev.vax.incremental.bahrain,4.87,True,2022-07-12T07:02:01,,
cowidev.vax.incremental.bangladesh,19.81,True,2022-07-12T07:02:18,,
cowidev.vax.incremental.barbados,11.18,True,2022-07-12T07:02:12,,
cowidev.vax.incremental.brazil,0.46,True,2022-07-12T07:02:09,,
cowidev.vax.incremental.bulgaria,4.62,True,2022-07-12T07:02:13,,
cowidev.vax.incremental.china,85.96,True,2022-07-12T07:05:54,,
cowidev.vax.incremental.costa_rica,,,2022-07-12T07:02:13,,
cowidev.vax.incremental.croatia,0.16,True,2022-07-12T07:02:13,,
cowidev.vax.incremental.cuba,2.37,True,2022-07-12T07:02:15,,
cowidev.vax.incremental.curacao,0.37,True,2022-07-12T07:02:14,,
cowidev.vax.incremental.cyprus,0.77,True,2022-07-12T07:02:14,,
cowidev.vax.incremental.dominican_republic,7.22,True,2022-07-12T07:02:22,,
cowidev.vax.incremental.el_salvador,1.11,True,2022-07-12T07:02:16,,
cowidev.vax.incremental.faeroe_islands,0.46,True,2022-07-12T07:02:17,,
cowidev.vax.incremental.fiji,0.0,True,2022-07-12T07:02:17,,
cowidev.vax.incremental.finland,0.24,True,2022-07-12T07:02:17,,
cowidev.vax.incremental.gabon,,,2022-07-12T07:02:17,,
cowidev.vax.incremental.georgia,0.67,True,2022-07-12T07:02:18,,
cowidev.vax.incremental.greenland,0.84,True,2022-07-12T07:02:19,,
cowidev.vax.incremental.guatemala,,,2022-07-12T07:02:18,,
cowidev.vax.incremental.guernsey,0.39,True,2022-07-12T07:02:19,,
cowidev.vax.incremental.hungary,1.33,True,2022-07-12T07:02:20,,
cowidev.vax.incremental.iceland,0.4,True,2022-07-12T07:02:19,,
cowidev.vax.incremental.india,1.19,True,2022-07-12T07:02:20,,
cowidev.vax.incremental.iran,2.12,True,2022-07-12T07:02:22,,
cowidev.vax.incremental.isle_of_man,0.37,True,2022-07-12T07:02:20,,
cowidev.vax.incremental.jamaica,1.77,True,2022-07-12T07:02:22,,
cowidev.vax.incremental.kazakhstan,10.04,True,2022-07-12T07:02:30,,
cowidev.vax.incremental.kosovo,1.91,True,2022-07-12T07:02:24,,
cowidev.vax.incremental.kyrgyzstan,1.03,True,2022-07-12T07:02:23,,
cowidev.vax.incremental.laos,2.1,True,2022-07-12T07:02:24,,
cowidev.vax.incremental.macao,21.7,False,2022-07-12T07:06:16,"Traceback (most recent call last):
  File ""/mnt/owid_live_covid/covid-19-data/scripts/src/cowidev/cmd/commons/get.py"", line 47, in run
    module.main()
  File ""/mnt/owid_live_covid/covid-19-data/scripts/src/cowidev/vax/incremental/macao.py"", line 84, in main
    Macao().export()
  File ""/mnt/owid_live_covid/covid-19-data/scripts/src/cowidev/vax/incremental/macao.py"", line 70, in export
    data = self.read()
  File ""/mnt/owid_live_covid/covid-19-data/scripts/src/cowidev/vax/incremental/macao.py"", line 20, in read
    elem = driver.find_element_by_partial_link_text(""Daily Bulletin on COVID-19"")
  File ""/mnt/owid_live_covid/covid-19-data/scripts/venv/lib/python3.9/site-packages/selenium/webdriver/remote/webdriver.py"", line 462, in find_element_by_partial_link_text
    return self.find_element(by=By.PARTIAL_LINK_TEXT, value=link_text)
  File ""/mnt/owid_live_covid/covid-19-data/scripts/venv/lib/python3.9/site-packages/selenium/webdriver/remote/webdriver.py"", line 976, in find_element
    return self.execute(Command.FIND_ELEMENT, {
  File ""/mnt/owid_live_covid/covid-19-data/scripts/venv/lib/python3.9/site-packages/selenium/webdriver/remote/webdriver.py"", line 321, in execute
    self.error_handler.check_response(response)
  File ""/mnt/owid_live_covid/covid-19-data/scripts/venv/lib/python3.9/site-packages/selenium/webdriver/remote/errorhandler.py"", line 242, in check_response
    raise exception_class(message, screen, stacktrace)
selenium.common.exceptions.NoSuchElementException: Message: no such element: Unable to locate element: {""method"":""partial link text"",""selector"":""Daily Bulletin on COVID-19""}
  (Session info: headless chrome=101.0.4951.64)

","Message: no such element: Unable to locate element: {""method"":""partial link text"",""selector"":""Daily Bulletin on COVID-19""}
  (Session info: headless chrome=101.0.4951.64)
"
cowidev.vax.incremental.moldova,0.92,True,2022-07-12T07:02:24,,
cowidev.vax.incremental.monaco,27.44,True,2022-07-12T07:02:52,,
cowidev.vax.incremental.mongolia,1.13,True,2022-07-12T07:02:26,,
cowidev.vax.incremental.montenegro,0.1,True,2022-07-12T07:02:26,,
cowidev.vax.incremental.morocco,,,2022-07-12T07:02:26,,
cowidev.vax.incremental.myanmar,,,2022-07-12T07:02:26,,
cowidev.vax.incremental.nepal,,,2022-07-12T07:02:26,,
cowidev.vax.incremental.north_macedonia,,,2022-07-12T07:02:26,,
cowidev.vax.incremental.northern_cyprus,0.83,True,2022-07-12T07:02:27,,
cowidev.vax.incremental.paho,51.45,True,2022-07-12T07:03:18,,
cowidev.vax.incremental.pakistan,1.34,True,2022-07-12T07:02:32,,
cowidev.vax.incremental.philippines,,,2022-07-12T07:02:32,,
cowidev.vax.incremental.poland,0.26,True,2022-07-12T07:02:32,,
cowidev.vax.incremental.qatar,,,2022-07-12T07:02:32,,
cowidev.vax.incremental.russia,0.96,False,2022-07-12T07:06:17,"Traceback (most recent call last):
  File ""/mnt/owid_live_covid/covid-19-data/scripts/src/cowidev/cmd/commons/get.py"", line 47, in run
    module.main()
  File ""/mnt/owid_live_covid/covid-19-data/scripts/src/cowidev/vax/incremental/russia.py"", line 64, in main
    data = read(source).pipe(pipeline)
  File ""/mnt/owid_live_covid/covid-19-data/scripts/src/cowidev/vax/incremental/russia.py"", line 11, in read
    soup = get_soup(source)
  File ""/mnt/owid_live_covid/covid-19-data/scripts/src/cowidev/utils/web/scraping.py"", line 83, in get_soup
    response = get_response(source, request_method, use_proxy, **kwargs)
  File ""/mnt/owid_live_covid/covid-19-data/scripts/src/cowidev/utils/web/scraping.py"", line 52, in get_response
    raise ValueError(
ValueError: Source https://gogov.ru/articles/covid-v-stats not reached! Error code 502 Bad Gateway
",Source https://gogov.ru/articles/covid-v-stats not reached! Error code 502 Bad Gateway
cowidev.vax.incremental.saint_lucia,,,2022-07-12T07:02:34,,
cowidev.vax.incremental.serbia,5.93,True,2022-07-12T07:02:40,,
cowidev.vax.incremental.south_africa,6.0,True,2022-07-12T07:02:46,,
cowidev.vax.incremental.spain,3.91,True,2022-07-12T07:02:49,,
cowidev.vax.incremental.sri_lanka,,,2022-07-12T07:02:46,,
cowidev.vax.incremental.suriname,16.12,True,2022-07-12T07:03:02,,
cowidev.vax.incremental.taiwan,8.17,True,2022-07-12T07:02:57,,
cowidev.vax.incremental.thailand,1.89,True,2022-07-12T07:02:54,,
cowidev.vax.incremental.turkey,0.53,True,2022-07-12T07:02:54,,
cowidev.vax.incremental.united_arab_emirates,,,2022-07-12T07:02:54,,
cowidev.vax.incremental.vietnam,,,2022-07-12T07:02:54,,
cowidev.vax.incremental.who,5.3,True,2022-07-12T07:02:59,,
cowidev.vax.incremental.zambia,0.97,True,2022-07-12T07:02:58,,
